RubeesEat/Model/DB/DbPersonRepository.cs

```diff
@@ -65,28 +84,65 @@ public void Add(Person person)
     
     public Person GetCurrentUser()
     {
-        return GetAll()[^1];
+        var httpContext = _httpContextAccessor.HttpContext;
+        if (httpContext == null)
+            throw new NullReferenceException("HttpContext is null");
+        var loginName = httpContext.User.FindFirst("name")?.Value;
+        var nickName = httpContext.User.FindFirst("nickname")?.Value;
```

Comment:
Yeah, the only reason I did it in this way, was to avoid having to change the function signature. But I can change it so that it passes the IPrinicipal instead. I think this is the only place where we use HttpContext in this way.

Reply:
From an outside perspective, this sounds like a sensible change. I can't speak to the implication for getting this done in time.